### PR TITLE
Feature/issue 537

### PR DESCRIPTION
**Pull request by cdausmus**
Updated query selecting general topics to SQL due to change on domain class requiring child objects loaded by join (and not lazy). Also fixed redirect after deleting a topic back to the thread.

### CODE REVIEW

File: grails-app/services/au/org/ala/volunteer/ForumService.groovy [hunk 1]
```diff
@@ -405,6 +400,9 @@ class ForumService {
             it.delete(flush: true)
         }
 
+        // Remove message from topic
+        ForumTopic forumTopic = message.topic
+        forumTopic.removeFromMessages(message)
```
Comment (temi) [inline grails-app/services/au/org/ala/volunteer/ForumService.groovy:405]: Does forumTopic need to be saved here?

**Review by temi — CHANGES_REQUESTED** [review]
added a comment

File: grails-app/services/au/org/ala/volunteer/ForumService.groovy (hunk 1 above)
Comment (cdausmus) [reply grails-app/services/au/org/ala/volunteer/ForumService.groovy:405]: Testing tells us no, but could add the save just to be sure?

**Comment by cdausmus** [comment]
Good thing you brought this up, found an issue that I hadn't tested for. Both have been fixed and tested.

**Review by temi — APPROVED** [review]

**Pull request merged**